### ISSUE TITLE
nix-prefetch-url retry hangs indefinitely on DNS host name resolution failure

### ISSUE THREAD

**Issue by chkno** [NixOS/nix#3338]
```
$ nix-prefetch-url http://no.such.domain.example.com/no-such-file
warning: unable to download 'http://no.such.domain.example.com/no-such-file': Couldn't resolve host name (6); retrying in 290 ms
downloading 'http://no.such.domain.example.com/no-such-file'
<hangs forever>
```

It works fine with `--option download-attempts 1`.

```
$ nix --version
nix (Nix) 2.3.2
```

**Comment by timor** [comment]
I can confirm the same behavior with the same `nix` version.  If I remember correctly, it should actually show multiple attempts with increasing retry delays.

**Comment by edolstra** [comment]
`-vvvvv` shows that Nix retries the request, but then curl stops processing it:
```
curl: Server doesn't support multiplex yet, wait
curl: No connections available.
```

Oddly it works with `--no-http2`.

**Comment by edolstra** [comment]
This appears to be fixed in the curl 7.67+.
